src/cose/decoder.rs

```diff
@@ -113,25 +102,21 @@ pub fn decode_signature(bytes: Vec<u8>, payload: &[u8]) -> Result<CoseSignatures
 
     // Parse the protected signature header.
     let protected_signature_header = decode(protected_signature_header_bytes.clone())?;
-    // This isn't quite right due to order... (it's a map)
-    let signature_algorithm = match protected_signature_header {
+    // TODO: This isn't quite right due to order... (it's a map)
+    let signature_algorithm_value = match protected_signature_header {
         CBORType::Map(values) => {
             if values.len() < 1 {
                 return Err("This is not a valid COSE signature object. Protected header is empty.");
             }
-            values[0].clone()
-        },
-        _ => return Err("Invalid COSE signature object: protected header is not a map."),
-    };
-    match signature_algorithm.key {
-        CBORType::Integer(val) => {
-            if val != 1 {
-                return Err("Invalid COSE signature: expected alg header key (1).");
+            let val = values.get(&CBORType::Integer(1)); //.unwrap().clone()
```

Comment:
Probably don't need the commented-out code, right?

Reply:
oops